src/Microsoft.AspNet.Mvc.TagHelpers/ScriptTagHelper.cs

```diff
@@ -0,0 +1,123 @@
+        [Activate]
+        protected internal ILogger<ScriptTagHelper> Logger { get; set; }
+
+        /// <inheritdoc />
+        public override async Task ProcessAsync(TagHelperContext context, TagHelperOutput output)
+        {
+            // TODO: Consider using the RazorStringWriter instead of a StringBuilder, because Script tags may contain large bodies.
+
+            if (!context.AllRequiredAttributesArePresent(RequiredAttributes, Logger))
+            {
+                if (Logger.IsEnabled(LogLevel.Verbose))
+                {
+                    Logger.WriteVerbose("Skipping processing for {0} {1}", nameof(ScriptTagHelper), context.UniqueId);
+                }
+
+                return;
+            }
+
+            var content = new StringBuilder();
+
+            // NOTE: Values in TagHelperOutput.Attributes are already HtmlEncoded
+
+            // We've taken over rendering here so prevent the element rendering the outer tag
+            output.TagName = null;
+
+            // Rebuild the <script /> tag.
+            content.Append("<script");
+            foreach (var attribute in output.Attributes)
+            {
+                content.AppendFormat(CultureInfo.InvariantCulture, " {0}=\"{1}\"", attribute.Key, attribute.Value);
+            }
+
+            content.Append(">");
+
+            var originalContent = await context.GetChildContentAsync();
+            content.Append(originalContent);
+            content.AppendLine("</script>");
+
+            // Build the <script /> tag that checks the test method and if it fails, renders the extra script.
+            content.Append("<script>(");
+            content.Append(FallbackTestExpression);
+            content.Append("||document.write(\"<script");
+
+            if (!output.Attributes.ContainsKey("src"))
+            {
+                AppendSrc(content, "src");
+            }
+
+            foreach (var attribute in output.Attributes)
+            {
+                if (!attribute.Key.Equals(SrcAttributeName, StringComparison.OrdinalIgnoreCase))
+                {
+                    var encodedKey = JavaScriptUtility.JavaScriptStringEncode(attribute.Key);
+                    var encodedValue = JavaScriptUtility.JavaScriptStringEncode(attribute.Value);
+
+                    content.AppendFormat(CultureInfo.InvariantCulture, " {0}=\\\"{1}\\\"", encodedKey, encodedValue);
+                }
+                else
+                {
+                    AppendSrc(content, attribute.Key);
+                }
+            }
+
+            content.Append("><\\/script>\"));</script>");
+
+            output.Content = content.ToString();
+        }
+
+        private void AppendSrc(StringBuilder content, string srcKey)
+        {
+            // Append src attribute in the original place and replace the content the the fallback content
+            // No need to encode the key because we know it is exactly src.
```

Comment:
`we know it is src` - Shouldn't say exactly because it's a case-insensitive comparison above.